plugins/filter/json_patch.yml

```diff
@@ -0,0 +1,140 @@
+---
+# Copyright (c) Stanislav Meduna (@numo68)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+DOCUMENTATION:
+  name: json_patch
+  short_description: Apply a JSON-Patch (RFC 6902) operation to an object
+  description:
+    - This filter applies a single JSON patch operation and returns a modified object.
+    - If the operation is a test, the filter returns an ummodified object if the test
+      succeeded and a None value otherwise.
+  requirements:
+    - jsonpatch
+  version_added: 10.3.0
+  author:
+    - Stanislav Meduna (@numo68)
+  positional: op, path, value
+  options:
+    _input:
+      description: A list or a dictionary representing a JSON object, or a string containing a JSON object.
+      type: raw
+      required: true
+    op:
+      description: Operation to perform (see RFC 6902).
+      type: str
+      choices: [add, copy, move, remove, replace, test]
```

Comment:
I think it would be useful if this filter and `json_patch_recipe` have an option which allows to turn failed `test`s into proper errors (`AnsibleFilterError` exception). Otherwise you'll always have to write extra code in the playbook to make sure that the result isn't `none` before using the result of these filters.

Reply:
There is, but only from inside another Jinja plugin and while using the 'native' engine. From Ansible itself you cannot pass byte strings, only unicode strings.